### ISSUE TITLE
Tests are not found in mill project

### ISSUE THREAD

**Issue by lefou** [lefou/LambdaTest#40]
After an experimental migration of a Java project from Maven (Polyglot Scala) to mill, LambdaTest test cases were ignored. Those tests were using the `testng.FreeSpec` base class.
 
Mill uses the `sbt.testing` abstraction with a dedicated adapter for TestNG `mill.testng.TestNGFramework`. See https://github.com/lihaoyi/mill/issues/553.

**Comment by lefou** [comment]
After explicitly overriding the annotated test method, the tests run as expected:

```java
  @Override
  @Test(dataProvider = "freeSpecTestCases")
  public void runFreeSpecTestCases(final DefaultTestCase testCase) throws Throwable {
    super.runFreeSpecTestCases(testCase);
  }
```

So, the issue is, that annotated methods are not searched in inherited classes.

**Comment by lefou** [comment]
It an issue in mill, which I already fixed in master. Should be available with next mill release > 0.3.6.
